json-schema-core/src/main/java/io/ballerina/jsonschema/core/Generator.java

```diff
@@ -0,0 +1,246 @@
+import static io.ballerina.jsonschema.core.GeneratorUtils.DECIMAL;
+import static io.ballerina.jsonschema.core.GeneratorUtils.FLOAT;
+import static io.ballerina.jsonschema.core.GeneratorUtils.INTEGER;
+import static io.ballerina.jsonschema.core.GeneratorUtils.JSON;
+import static io.ballerina.jsonschema.core.GeneratorUtils.NEVER;
+import static io.ballerina.jsonschema.core.GeneratorUtils.NULL;
+import static io.ballerina.jsonschema.core.GeneratorUtils.NUMBER;
+import static io.ballerina.jsonschema.core.GeneratorUtils.PIPE;
+import static io.ballerina.jsonschema.core.GeneratorUtils.PUBLIC;
+import static io.ballerina.jsonschema.core.GeneratorUtils.SEMI_COLON;
+import static io.ballerina.jsonschema.core.GeneratorUtils.STRING;
+import static io.ballerina.jsonschema.core.GeneratorUtils.TYPE;
+import static io.ballerina.jsonschema.core.GeneratorUtils.UNIVERSAL_ARRAY;
+import static io.ballerina.jsonschema.core.GeneratorUtils.UNIVERSAL_OBJECT;
+import static io.ballerina.jsonschema.core.GeneratorUtils.WHITE_SPACE;
+import static io.ballerina.jsonschema.core.GeneratorUtils.IMPORT;
+import static io.ballerina.jsonschema.core.GeneratorUtils.createType;
+import static io.ballerina.jsonschema.core.SchemaUtils.ID_TO_TYPE_MAP;
+
+public class Generator {
+    static final String DEFAULT_SCHEMA_NAME = "Schema";
+    static final String EOF_TOKEN = "";
+    static final String INVALID_IMPORTS_ERROR = "Invalid imports have been found.";
+
+    Map<String, ModuleMemberDeclarationNode> nodes = new LinkedHashMap<>();
+    ArrayList<String> imports = new ArrayList<>();
+    List<JsonSchemaDiagnostic> diagnostics = new ArrayList<>();
+
+    public Response convertBaseSchema(Object schemaObject) throws Exception {
+        String generatedType = convert(schemaObject, DEFAULT_SCHEMA_NAME);
+
+        if (!generatedType.equals(DEFAULT_SCHEMA_NAME)) {
+            String schemaDefinition = PUBLIC + WHITE_SPACE + TYPE + WHITE_SPACE
+                    + DEFAULT_SCHEMA_NAME + WHITE_SPACE + generatedType + SEMI_COLON;
+            ModuleMemberDeclarationNode schemaNode = NodeParser.parseModuleMemberDeclaration(schemaDefinition);
+            this.nodes.put(DEFAULT_SCHEMA_NAME, schemaNode);
+        }
+
+        ModulePartNode modulePartNode = generateModulePartNode();
+        String generatedTypes = formatModuleParts(modulePartNode);
+        return new Response(generatedTypes, this.diagnostics);
+    }
+
+    public String convert(Object schemaObject, String name) {
+        if (schemaObject instanceof Boolean boolValue) {
+            return boolValue ? JSON : NEVER;
+        }
+
+        Schema schema = (Schema) schemaObject;
+
+        List<Object> schemaType = getCommonType(schema.getEnumKeyword(), schema.getConstKeyword(),
+                schema.getType());
+        if (schemaType.isEmpty()) {
+            ID_TO_TYPE_MAP.put(schema.getIdKeyword(), NEVER);
+            return NEVER;
+        } else if (schemaType.contains(Class.class)) {
```

Comment:
While this exact code will change, we prefer using the early return pattern.

https://medium.com/swlh/return-early-pattern-3d18a41bba8